Shopify/cli
# [Refactor] Move HostThemeManager and HostThemeManager into themes app extensions utilities directory

### WHY are these changes introduced?
- Improves code organization by grouping code related to theme app extensions
- Closes https://github.com/Shopify/cli/issues/4328

### WHAT is this pull request doing?
1. Move `host-theme-manager` and `host-theme-watcher` under the `theme` folder for app extensions
2. Update imports and test mocks

### How to test your changes?

Run app package tests
- `cd packages/app && pnpm test`

Try creating a Host Theme (first ensure that your host theme isn't already present)
- This will require having an initialized app with an app extension

![image](https://github.com/user-attachments/assets/7612e8b4-77f4-4b6f-bbd8-145cf25c931a)


### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
